src/components/ProjectIconSelection/ProjectIconSelection.js

```diff
@@ -0,0 +1,86 @@
+// @flow
+import React, { Component } from 'react';
+import styled from 'styled-components';
+import importAll from 'import-all.macro';
+
+import { sampleMany } from '../../utils';
+import { BREAKPOINTS } from '../../constants';
+
+import SelectableImage from '../SelectableImage';
+
+const icons: Array<mixed> = importAll.sync(
+  '../../assets/images/icons/icon_*.*'
+);
+const iconSrcs: Array<string> = Object.values(icons).map(src => String(src));
+
+const DEFAULT_SUBSET = 10;
+const DEFAULT_ICON_SIZE = 60;
+
+type Props = {
+  selectedIcon: ?string,
+  showRandomSubset: boolean | number,
+  onSelectIcon: (src: string, ev: SyntheticMouseEvent<*>) => void,
+};
+
+type State = {
+  selectableIcons: Array<string>,
+};
+
+class ProjectIconSelection extends Component<Props, State> {
+  state = {
+    selectableIcons: [],
+  };
+  componentDidMount() {
+    const { showRandomSubset } = this.props;
+    const selectableIcons = !showRandomSubset
+      ? iconSrcs
+      : typeof showRandomSubset === 'number'
+        ? sampleMany(iconSrcs, showRandomSubset)
+        : sampleMany(iconSrcs, DEFAULT_SUBSET);
+
+    this.setState({
+      selectableIcons,
+    });
+  }
+  render() {
+    const { selectedIcon } = this.props;
+    const { selectableIcons } = this.state;
+
+    return (
+      <ProjectIconWrapper>
+        {selectableIcons.map((src: string) => (
+          <SelectableImageWrapper key={src}>
+            <SelectableImage
+              src={src}
+              size={DEFAULT_ICON_SIZE}
+              onClick={ev => this.props.onSelectIcon(src, ev)}
+              status={
+                selectedIcon === null
+                  ? 'default'
+                  : selectedIcon === src
+                    ? 'highlighted'
+                    : 'faded'
+              }
+            />
+          </SelectableImageWrapper>
+        ))}
+      </ProjectIconWrapper>
+    );
+  }
+}
+
+const ProjectIconWrapper = styled.div`
+  margin-top: 16px;
+`;
+
+const SelectableImageWrapper = styled.div`
+  display: inline-block;
+  margin: 0px 10px 10px 0px;
+  @media ${BREAKPOINTS.sm} {
+    &:nth-of-type(n + 9) {
+      display: none;
+    }
+  }
```

Comment:
Ah, so this breakpoint is a bit of an issue; we don't want to trim the subset when it's within `ProjectSettingsModal`. This could lead to the situation where the selected item isn't displayed. Plus we don't need to trim it, it can fit.

One possible solution could be to make the `limitTo` prop take an object, with different numbers for different breakpoints... but I really don't like that idea, it feels complex and troublesome.

Instead, if we had a way to gauge the breakpoint at the component level, we could do something like this:

```js
// in MainPane.js

<CurrentBreakpoint>
  {(size) => (
    <ProjectIconSelection
      limitTo={size === "sm" ? 8 : 10}
    />
  )}
</CurrentBreakpoint>
```

We don't have that `CurrentBreakpoint` component, though. For now I think it's probably OK to just set the MainPane value to 8 instead of 10, on all breakpoints. Not a big deal to show a couple less, to ensure it always works.

Reply:
As discussed in the general comments, I think we should remove this block: 

```
  @media ${BREAKPOINTS.sm} {
    &:nth-of-type(n + 9) {
      display: none;
    }
  }
```